### DISCUSSION TITLE
Four displays on an ESP32

### DISCUSSION THREAD

**Discussion by Scugnizzo** [Bodmer/TFT_eSPI#3004] [Q&A - General]
Hi Bodmer and thanks for this useful library.
I want do build a clock with an ESP32, four ST7735 0.96 inch SPI displays and a SPI SD card reader.
I know that each HSPI and VSPI bus can handle up to three devices.
It's possible to run all my five SPI devices without an SPI multiplexer ?
Thank you for your answer.

**Comment by Bodmer** [comment]
This may help:
https://github.com/kitecraft/Multiple_TFT_Displays
https://www.youtube.com/watch?v=cCgNHIHijhs

The CS line must be controlled in the sketch and the displays need to all use the same driver.

**Reply by Scugnizzo** [reply]
Thank you for your reply Bodmer,
as you suggested, I leave commented the TFT_CS line in the User_Setup.h file and control two different CS lines in the sketch.
I've tried only one simple command to test the displays (ST7735) but the displays continue to remain black.
if I manually connect one CS line at a time to GND, the displays works fine (obviously they display all the data received on the  SPI bus).
 
Here my sketch:

#include <SPI.h>
#include <FS.h>
#include <SD.h>
#include <TFT_eSPI.h>
TFT_eSPI tft = TFT_eSPI();
#define TFT3_CS  21 // CS first select line
#define TFT2_CS  22 // CS second select line

void setup() {
Serial.begin(115200);
pinMode(TFT2_CS, OUTPUT);
pinMode(TFT3_CS, OUTPUT);
// Set all chip selects high to avoid bus contention during initialisation of each peripheral
digitalWrite(TFT3_CS, HIGH);
digitalWrite(TFT2_CS, HIGH); 
digitalWrite( 5, HIGH); // SD card chips select, must use GPIO 5 (ESP32 SS)
tft.begin();

  if (!SD.begin(5, tft.getSPIinstance())) {
    Serial.println("Card Mount Failed");
    return;
  }
  uint8_t cardType = SD.cardType();

  if (cardType == CARD_NONE) {
    Serial.println("No SD card attached");
    return;
  }

  Serial.println("initialisation done.");
}

void loop() {

  digitalWrite(TFT2_CS, LOW);
  digitalWrite(TFT3_CS, LOW);
  tft.setRotation(0);  // set rotation to both displays
  digitalWrite(TFT2_CS, HIGH);
  digitalWrite(TFT3_CS, HIGH);


digitalWrite(TFT2_CS, LOW);
tft.fillScreen(TFT_GREEN);  // fill the first display with green color
digitalWrite(TFT2_CS, HIGH);

delay(5000);

digitalWrite(TFT3_CS, LOW);
tft.fillScreen(TFT_RED);  // fill the second display with red color
digitalWrite(TFT3_CS, HIGH);

while (1);
 
}

Any help would be appreciated

**Reply by Scugnizzo** [reply]
better to use tft.init() instead of tft.begin(). Now it's all OK

**Reply by Bodmer** [reply]
You must set the two chip selects low before tft.init/begin so the driver cips get configured. Set high afterwards and then control each display individually.
